h2/src/test/org/h2/test/db/TestLocalResultFactory.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2004-2018 H2 Group. Multiple-Licensed under the MPL 2.0,
+ * and the EPL 1.0 (http://h2database.com/html/license.html).
+ * Initial Developer: H2 Group
+ */
+package org.h2.test.db;
+
+import java.sql.Connection;
+import java.sql.Statement;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.h2.engine.Session;
+import org.h2.expression.Expression;
+import org.h2.result.LocalResult;
+import org.h2.result.LocalResultFactory;
+import org.h2.test.TestBase;
+import org.h2.test.TestDb;
+
+/**
+ * Test {@link LocalResultFactory} setting.
+ */
+public class TestLocalResultFactory extends TestDb {
```

Comment:
Do you have any reason to test it in all modes?

From my point of view it may extend `TestBase` and be tested only one time with some in-memory database.

(In this case it should be moved to a better package, like `test.unit` and be called from `TestAll.testUtils()` instead.)